### PR TITLE
apple/apple3_v.cpp: Add support for flashing chars and remove smooths…

### PR DESCRIPTION
**Pull request by robjustice**
…cr from text modes

### CODE REVIEW

File: src/mame/apple/apple3_v.cpp [hunk 1]
```diff
@@ -167,33 +166,25 @@ void apple3_state::text40(bitmap_ind16 &bitmap, const rectangle &cliprect)
...
 			{
 				for (int col = 0; col < 7; col++)
 				{
-					int lc;
-					if (m_smoothscr)
+					uint16_t *const dest = &bitmap.pix(y * 8 + row, x * 14 + col * 2);
+
+					// Flash or Inverse
+					if ((!(ch & 0x80) && (char_data[row] & 0x80) && m_flash) || (!(ch & 0x80) && !(char_data[row] & 0x80)))
 					{
```
Comment (cuavas) [inline src/mame/apple/apple3_v.cpp:179]: This seems to have excess parentheses and common subexpressions.  Wouldn’t it be clearer as:
```c++
 		if (!(ch & 0x80) && (!(char_data[row] & 0x80) || m_flash))
```

That’s equivalent logic after minimising the truth table.

File: src/mame/apple/apple3_v.cpp [hunk 2]
```diff
@@ -219,44 +209,50 @@ void apple3_state::text80(bitmap_ind16 &bitmap, const rectangle &cliprect)
...
 			{
 				for (int col = 0; col < 7; col++)
 				{
-					int lc;
-					if (m_smoothscr)
+					uint16_t *const dest = &bitmap.pix(y * 8 + row, x * 14 + col + 0);
+
+					// Flash or Inverse
+					if ((!(ch & 0x80) && (char_data[row] & 0x80) && m_flash) || (!(ch & 0x80) && !(char_data[row] & 0x80)))
 					{
```
Comment (cuavas) [inline src/mame/apple/apple3_v.cpp:227]: Same here – it can use the same minimised expression.

File: src/mame/apple/apple3_v.cpp [hunk 3]
```diff
@@ -219,44 +209,50 @@ void apple3_state::text80(bitmap_ind16 &bitmap, const rectangle &cliprect)
...
 			for (int row = 0; row < 8; row++)
 			{
 				for (int col = 0; col < 7; col++)
 				{
 					uint16_t *const dest = &bitmap.pix(y * 8 + row, x * 14 + col + 7);
-					*dest = (char_data[row] & (1 << col)) ? fg : bg;
+
+					// Flash or Inverse
+					if ((!(ch & 0x80) && (char_data[row] & 0x80) && m_flash) || (!(ch & 0x80) && !(char_data[row] & 0x80)))
+					{
```
Comment (cuavas) [inline src/mame/apple/apple3_v.cpp:249]: And this one as well.

**Review by cuavas — CHANGES_REQUESTED** [review]

File: src/mame/apple/apple3_v.cpp (hunk 1 above)
Comment (robjustice) [reply src/mame/apple/apple3_v.cpp:179]: yes, that's much better.

Comment (robjustice) [reply src/mame/apple/apple3_v.cpp]: updated

File: src/mame/apple/apple3_v.cpp (hunk 3 above)
Comment (robjustice) [reply src/mame/apple/apple3_v.cpp]: updated this as well

File: src/mame/apple/apple3_v.cpp (hunk 2 above)
Comment (robjustice) [reply src/mame/apple/apple3_v.cpp]: updated

**Pull request merged**